include/megaapi_impl.h

```diff
@@ -2596,9 +2596,9 @@ class MegaApiImpl : public MegaApp
         void getCountryCallingCodes(MegaRequestListener *listener = NULL);
 
         void fireOnTransferStart(MegaTransferPrivate *transfer);
-        void fireOnTransferFinish(MegaTransferPrivate *transfer, MegaError *e, DBTableTransactionCommitter& committer);
+        void fireOnTransferFinish(MegaTransferPrivate *transfer, unique_ptr<MegaError> e, DBTableTransactionCommitter& committer);
```

Comment:
I think we should pass the `MegaErrorPrivate` here, instead of `MegaError`.
Same for `fireOnTransferTemporaryError()`, `fireOnBackupFinish()` and `fireOnBackupTemporaryError()`